simulation/eor_bubble_sim.pro

```diff
@@ -0,0 +1,84 @@
+FUNCTION eor_bubble_sim, obs, jones, select_radius=select_radius, bubble_fname=bubble_fname, beam_threshold=beam_threshold, allow_sidelobe_sources=allow_sidelobe_sources, dat=dat 
+
+;Opening an HDF5 file and extract relevant data
+if keyword_set(bubble_fname) THEN hdf5_fname = bubble_fname ELSE hdf5_fname = '/users/alanman/data/alanman/BubbleCube/TiledHpxCubes/mwa_gaussian.hdf5'
+if not keyword_set(beam_threshold) then beam_threshold = 0.05
+if keyword_set(allow_sidelobe_sources) THEN beam_threshold = 0.01
+if not keyword_set(select_radius) THEN  select_radius = 20     ; Degrees
+
+dimension=obs.dimension
+elements= obs.elements
+
+f_id = H5F_OPEN(hdf5_fname)
+dset_id_eor  = H5D_OPEN(f_id, '/spectral_info/spectrum')
+dspace_id_eor = H5D_GET_SPACE(dset_id_eor)
+
+freq_hpx = H5_GETDATA(hdf5_fname, '/spectral_info/freq')
+ra_hpx = H5_GETDATA(hdf5_fname, '/object/RA') 
+dec_hpx = H5_GETDATA(hdf5_fname, '/object/Dec') 
+
+dims = REVERSE(H5S_GET_SIMPLE_EXTENT_DIMS(dspace_id_eor))
+nside=NPIX2NSIDE(dims[0])
+nfreq_hpx = dims[1]
+
+n_pol=obs.n_pol
+
+; Identify the healpix pixels within select_radius of the primary beam
+ang2vec,obs.obsdec,obs.obsra,cen_coords,/astro
+Query_disc,nside,cen_coords,select_radius,inds_select,npix_sel,/deg
+print, 'selection radius (degrees) ', select_radius
+print, "Npix_selected: ", npix_sel
+
+; Limit the range of frequencies in the uvf cube to the range of the obs
+freq_arr = (*obs.baseline_info).freq
+lim = minmax(freq_arr)
+freq_inds = where((freq_hpx GT lim[0]) and (freq_hpx LT lim[1]) )
+freq_hpx = freq_hpx[freq_inds]
+nfreq_hpx = n_elements(freq_hpx)
+
+;; Extract only these healpix indices from the file.
+;H5S_SELECT_ELEMENTS, dspace_id_eor, hpx_inds, /RESET
+
+print, "Reading HDF5 file with EoR Healpix Cube"
+t0 = systime(/seconds)
+if n_elements(dat) eq 0 then dat = H5D_READ(dset_id_eor, FILE_SPACE=dpsace_id_eor)  ; temporary --- allow for the data to be reused if already loaded
```

Comment:
Comment says this is temporary?

Reply:
It doesn't need to be temporary... I was doing some tests and made it so eor_bubble_sim could accept the data object instead of re-reading it to save on time. It's not currently used, but might be useful. I can remove the option if you'd prefer.